### ISSUE TITLE
Optimize G only for latent L1 loss

### ISSUE THREAD

**Issue by FishYuLi** [junyanz/BicycleGAN#5]
https://github.com/junyanz/BicycleGAN/blob/5bd76aca5ed3d9e62aebb578a28a1053b622ddf9/models/bicycle_gan_model.py#L161

Hi, Junyan,
I am confused why the latent L1 loss is only used to optimize G only, but not used to optimize E?

By the way [this line](https://github.com/junyanz/BicycleGAN/blob/5bd76aca5ed3d9e62aebb578a28a1053b622ddf9/models/bicycle_gan_model.py#L58) 
` self.fake_data_random = torch.cat([self.real_A_encoded, self.fake_B_random], 1)` 
should be
 `self.fake_data_random = torch.cat([self.real_A_random, self.fake_B_random], 1)`.
Right?

**Comment by junyanz** [comment]
Thanks for your questions. 
1. In Line 161, we only update G for the latent L1 loss while the E is fixed. We found optimizing G and E simultaneously will encourage G and E to hide the information of the latent vector without learning meaningful styles. We will clarify this detail in our revision. 
2. This [line](https://github.com/junyanz/BicycleGAN/blob/5bd76aca5ed3d9e62aebb578a28a1053b622ddf9/models/bicycle_gan_model.py#L58) is correct. We generate fake_B_random from real_A_encoded as shown in this [line](https://github.com/junyanz/BicycleGAN/blob/5bd76aca5ed3d9e62aebb578a28a1053b622ddf9/models/bicycle_gan_model.py#L54). Sorry for the naming confusion. Also in the default setting,
 opt.conditional_D=False.
